### DISCUSSION TITLE
HLS AUTH

### DISCUSSION THREAD

**Discussion by pikachu937** [bluenviron/mediamtx#1759] [Questions]
Greetings to all.
Once again I express my respect to aler9, you are the best.
I've run into a problem and I'm asking for your help.
We have mediamtx-0.22.2 and 10 rtsp streams. 
External authentication is configured, all clients are safely authorized to view rtsp streams using a link like: **rtsp://login:password@ip:port/stream**(if you specify an incorrect login and password or do not specify access to the stream at all, therefore, you can do conclusion that authentication is working).).
I turned on LL-HLS and try to connect to the stream via the safari/chrome/firefox browser using the link **https://login:password@ip:port/stream/**, but for some reason the browser asks me to re-enter my login and password in the form. I also tried using xcode to write a simple player to view ll-hls on my iphone, the result is the same, I get an error that I am not authorized.
I did not lose heart and checked the availability of input data through curl and, lo and behold, curl returned the html page to me, and in the mediamtx logs I saw the treasured lines:
2023/05/03 21:15:54 DEB [HLS] [conn 192.168.1.77:50783] GET /path1/
2023/05/03 21:15:54 DEB [HLS] [conn 192.168.1.77:50783] [c->s] GET /path1/ HTTP/2.0
Host: 192.168.1.150:8888
Accept: */*
Authorization: Basic 
uybtiuytiITBnuygYGBrvytRYTrvy5tr55HJiuyn7FDFX7687ttvucytetyrexCTRCFVHGFBJHGJNJKHgdcGFdhgv8786fuyvViITBfstz643gfhxjgfxyetwt4ee65ifghcFXGXGFXKtr563w
User Agent: curl/7.86.0

2023/05/03 21:15:54 INF [HLS] [muxer path1] created (requested by 192.168.1.77)
2023/05/03 21:15:54 INF [HLS] [muxer path1] is converting into HLS, 1 track (H264)
2023/05/03 21:15:54 DEB [HLS] [conn 192.168.1.77:50783] [s->c] HTTP/1.1 200 OK
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: *
Content-Type: text/html
Server: mediamtx

The question is how to generate a url so that the browser accepts the data entered in the address without asking for a re-entry of the login and password, and how can I connect ll-hls in the application on the phone.
Thank you.

PS. By the way, looking for similar problems in discuss and issue, I found one of them [return www-authenticate with 401](https://github.com/aler9/mediamtx/discussions/1117), possible regression.

UPD. Got it to work on iphone and xiaomi in my apps but in browser i still can't access via https://login:password@ip:port/stream/

**Comment by aler9** [comment]
answered in #1777
